### ISSUE TITLE
When can we expect full GUI Support

### ISSUE THREAD

**Issue by TheDingOfTheHud** [darlinghq/darling#1069]
Title, don't really need to say much more
(Also any apps that currently have fully GUI support? Hoping to run Logic Pro X, Final Cut Pro & Xcode)

**Comment by aronkvh** [comment]
Related to this: Is there a DarlingDB like ProtonDB, to see which apps can run for other users?

**Comment by Etaash-mathamsetty** [comment]
> Title, don't really need to say much more (Also any apps that currently have fully GUI support? Hoping to run Logic Pro X, Final Cut Pro & Xcode)

that will take a long time, since some terminal applications still don't run correctly (missing libraries, etc)

**Comment by Etaash-mathamsetty** [comment]
The thing I don't understand is why wasn't mac os first,
why did wine start first and not a project like darling since mac os is so similar to linux

**Comment by TheDingOfTheHud** [comment]
Windows is a more popular OS thus there's more apps that people want to use
off windows

**Comment by Wind-Explorer** [comment]
What Darling is aiming at is basically rewriting macOS without looking at its source code. It took Wine 28 years to do the same thing with Windows since the year 1993 and got to where it is today, still with a ton of bugs and issues. Therefore I suspect that GUI support will most likely take more than 5-6 years since Darling was first started in 2012. I would say 10 to 20 years would be enough-ish for this project to become useful for ones trying to do actually productive work and have a genuine macOS-like experience without major issues with this compatibility layer.

**Comment by Etaash-mathamsetty** [comment]
> What Darling is aiming at is basically rewriting macOS without looking at its source code. It took Wine 28 years to do the same thing with Windows since the year 1993 and got to where it is today, still with a ton of bugs and issues. Therefore I suspect that GUI support will most likely take more than 5-6 years since Darling was first started in 2012. I would say 10 to 20 years would be enough-ish for this project to become useful for ones trying to do actually productive work and have a genuine macOS-like experience without major issues with this compatibility layer.

yeah maybe 10 years, but the wine devs had to reverse engineer EVERYTHING. Apple has some open source stuff like the kernel, so it's much easier. MacOS is also unix based just like linux, which makes it easier. I would say the challenges are Apple's proprietary libraries, the UI, and metal (graphics/compute api).

**Comment by HinTak** [comment]
On the other hand, executor, pearpc, sheepshaver etc never took on. There are "important" windows applications, like Microsoft Office, that one wants to run on linux. Try naming one mac application that has difficult user bases that one wants to run elsewhere. Besides, the user base of mac itself is smaller.

**Comment by HinTak** [comment]
And technology aside, crossover is actually a well-run company with smart business people, good cash flow, and an approachable, helpful and professional dev team, from the beginning... Darling does not have any of that. Far from it.

**Comment by TheBrokenRail** [comment]
> helpful and professional dev team, from the beginning... Darling does not have any of that. Far from it.

Really, personal attacks now? That's just sad.

**Comment by HinTak** [comment]
That's a good example of being unhelpful and unprofessional. Thanks for the demonstration.

**Comment by rin67630** [comment]
> What Darling is aiming at is basically rewriting macOS without looking at its source code. 
On the top of that Apple permanently changes macOS APIs jeopardizing older apps, what Microsoft does not do, carefully ensuring backwards compatibility.
It's practically impossible to keep on pace.

**Comment by Etaash-mathamsetty** [comment]
> > What Darling is aiming at is basically rewriting macOS without looking at its source code.
> 
> On the top of that Apple permanently changes macOS APIs jeopardizing older apps, what Microsoft does not do, carefully ensuring backwards compatibility. It's practically impossible to keep on pace.

we can just stick to one version and get it in a good state before updating...

**Comment by acheong08** [comment]
> On the other hand, executor, pearpc, sheepshaver etc never took on. There are "important" windows applications, like Microsoft Office, that one wants to run on linux. Try naming one mac application that has sufficient user bases that one wants to run elsewhere. Besides, the user base of mac itself is smaller.

iMessage is a pretty important application I suppose...

**Comment by stuaxo** [comment]
> The thing I don't understand is why wasn't mac os first, why did wine start first and not a project like darling since mac os is so similar to linux

Wine was released in 1993 [1], the first public beta of OSX was released in September 2000 [2]

[1] from the Wine_Software history on wikipedia
> Bob Amstadt, the initial project leader, and Eric Youngdale started the Wine project in **1993** as a way to run Windows applications on [Linux](https://en.wikipedia.org/wiki/Linux). It was inspired by two [Sun Microsystems](https://en.wikipedia.org/wiki/Sun_Microsystems)' products, the [Wabi](https://en.wikipedia.org/wiki/Wabi_(software)) for the [Solaris](https://en.wikipedia.org/wiki/Solaris_(operating_system)) operating system, and the [Public Windows Initiative](https://en.wikipedia.org/wiki/Application_Programming_Interface_for_Windows#ECMA_gets_involved),[[24]](https://en.wikipedia.org/wiki/Wine_(software)#cite_note-24) ...

[2] From the OSX history section wikipedia
> On **September 13, 2000**, Apple released a $29.95[[33]](https://en.wikipedia.org/wiki/MacOS_version_history#cite_note-33) "preview" version of [Mac OS X](https://en.wikipedia.org/wiki/Mac_OS_X_Public_Beta) (internally codenamed [Kodiak](https://en.wikipedia.org/wiki/Kodiak_bear))

**Comment by Etaash-mathamsetty** [comment]
> > The thing I don't understand is why wasn't mac os first, why did wine start first and not a project like darling since mac os is so similar to linux
> 
> Wine was released in 1993 [1], the first public beta of OSX was released in September 2000 [2]
> 
> [1] from the Wine_Software history on Wikipedia:
> > Bob Amstadt, the initial project leader, and Eric Youngdale started the Wine project in **1993** as a way to run Windows applications on [Linux](https://en.wikipedia.org/wiki/Linux). It was inspired by two [Sun Microsystems](https://en.wikipedia.org/wiki/Sun_Microsystems)' products, the [Wabi](https://en.wikipedia.org/wiki/Wabi_(software)) for the [Solaris](https://en.wikipedia.org/wiki/Solaris_(operating_system)) operating system, and the [Public Windows Initiative](https://en.wikipedia.org/wiki/Application_Programming_Interface_for_Windows#ECMA_gets_involved),[[24]](https://en.wikipedia.org/wiki/Wine_(software)#cite_note-24) ...
> 
> [2] From the Mac_OS_X_Public_Beta history section Wikipedia:
> > On **September 13, 2000**, Apple released a $29.95[[33]](https://en.wikipedia.org/wiki/MacOS_version_history#cite_note-33) "preview" version of [Mac OS X](https://en.wikipedia.org/wiki/Mac_OS_X_Public_Beta) (internally codenamed [Kodiak](https://en.wikipedia.org/wiki/Kodiak_bear))

I hope you realize that there were versions of Mac Os before Mac Os X

**Comment by stuaxo** [comment]
Of course, yes.

In that case, there was Executor which is the closest to something like Wine.

It implemented up to the equivalent of about System 6, and could run the software it could run very performantly on a 486.

It was closed source, but was since opened and can be found on GitHub.

More recent work has ported it to to cmake and 64 bit, it's certainly a fun project.
